utils/swift_build_support/swift_build_support/products/swiftpm.py

```diff
@@ -47,7 +47,10 @@ def run_bootstrap_script(self, action, host_target, additional_params=[]):
             self.source_dir, 'Utilities', 'bootstrap')
 
         toolchain_path = self.native_toolchain_path(host_target)
-        swiftc = os.path.join(toolchain_path, "bin", "swiftc")
+        if self.toolchain.swiftc is not None:
+            swiftc = self.toolchain.swiftc
```

Comment:
We should switch over Python build products to always use this line in subsequent pulls, rather than doing the dance two lines below of adding and removing directories manually.

Reply:
First iteration, I did try going down the route of just using `swift.toolchain.swiftc`, but the Linux toolchain doesn't have a swiftc by default so we do need it to fall back on the `return self.install_toolchain_path(host_target)` in `native_toolchain_path` for normal builds.
Unfortunately, toolchains don't necessarily have a single root path, so I don't really know what the short-term right answer is.
Long-term, it seems like we need to discuss a refactor so that we don't need the `native_toolchain_path` at all since toolchains don't necessarily have one, and can pull what we need from `self.toolchain`, or fall back on the installed compiler (assuming it wasn't a Canadian cross compile).